### DISCUSSION TITLE
Cron Expression Days of Week - Why 1-7?

### DISCUSSION THREAD

**Discussion by ghost** [quartznet/quartznet#1720] [Q&A] [answered]
Hi, I've found a couple of projects in github that translate cron expressions into a human readable format.
Both of them treat Sunday as 0 or as both 0 and 7. Quartz.Net uses Sunday = 1

Question: Why doesn't Quartz.Net follow that, since it seems to be the standard (based on many other sites / posts that explain cron syntax)?

I mainly ask because if I use CronScheduleBuilder.AtHourAndMinuteOnGivenDaysOfWeek to create a schedule, it uses numbers instead of MON, TUE, etc... unless there is a way to tell it to use the day abbreviation strings instead of the numbers.

This makes that method and .DailyAtHourAndMinute pretty much useless to us, which is a bummer because those two methods would have handled 95% of our needs for creating Cron schedules. Now I imagine we'll have to write the code to generate the expressions, which isn't a huge deal, but I'd love to have been able to use the methods...

Brent

**Comment by lahma** [comment, marked as answer]
The logic was ported from Java originally.

**Reply by ghost** [reply]
Thanks for the answer, and I understand. Just seems like cron has been around a long time so it is strange to have different standards out there, for something like this.

**Reply by lahma** [reply]
Cron also has precision to the minute, Quartz handles also second so format differs.
